src/PostgrestFilterBuilder.ts

```diff
@@ -25,6 +25,30 @@ type FilterOperator =
   | 'phfts'
   | 'wfts'
 
+// Match relationship filters with `table.column` syntax and resolve underlying column value.
+// If not matched, fallback to generic type.
+type ResolveFilterValue<
```

Comment:
Might have to make this recursive to account for deeply nested embedded tables, e.g. `.eq('messages.channels.id', 42)`

Reply:
Resolved in https://github.com/supabase/postgrest-js/pull/589/commits/697cc601d3eb31496b95e86bd705d5e6e861558b

Reply:
<3